cheatsheets/gleam-for-elixir-users.md

```diff
@@ -0,0 +1,235 @@
+---
+layout: page
+title: Gleam for Elixir users
+---
+
+- [Comments](#comments)
+- [Variables](#variables)
+  - [Match operator](#match-operator)
+  - [Variable type annotations](#variable-type-annotations)
+- [Functions](#functions)
+  - [Exporting functions](#exporting-functions)
+  - [Function type annotations](#function-type-annotations)
+  - [Function heads](#function-heads)
+  - [Function overloading](#function-overloading)
+  - [Referencing functions](#referencing-function)
+  - [Labelled arguments](#labelled-arguments)
+- [Operators](#operators)
+  - [Pipe](#pipe)
+- [Constants](#constants)
+- [Blocks](#blocks)
+- [Data types](#data-types)
+  - [Strings](#strings)
+  - [Tuples](#tuples)
+  - [Lists](#lists)
+  - [Atoms](#atoms)
+  - [Maps](#maps)
+- [Type aliases](#type-aliases)
+- [Custom types](#custom-types)
+  - [Records](#records)
+  - [Unions](#unions) TODO
+  - [Opaque custom types](#opaque-custom-types) TODO
+- [Flow control](#flow-control) TODO
+  - [Case](#case) TODO
+  - [Try](#try) TODO
+- [Modules](#modules) TODO
+  - [Imports](#imports) TODO
+  - [Nested modules](#nested-modules) TODO
+  - [First class modules](#first-class-modules) TODO
+
+## Variables
+
+#### Elixir
+
+In Elixir and Gleam are similar, only gleam has the `let` keyword before its variable names.
+You can reassign variables in both languages.
+
+```elixir
+size = 50
+size2 = size + 100
+size2 = 1
+```
+
+#### Gleam
+
+```rust
+let size = 50
+let size = size + 100
+let size = 1
+```
+
+### Match operator
+
+#### Elixir
+
+In Elixir `=` is really just a match operator and can be
+used to assert that a given term has a specific shape or value.
+
+```elixir
+[x] = [1] # assert `x` is a 1 element list and assign it to this element's value
+2 = x # error because x's value is 1
+```
+
+#### Gleam
+
+In Gleam `=` could be used for pattern matching as well, but you'll get compile errors if there's a type mismatch
+and a runtime error if there's a value mismatch. Also, there's the `assert` keyword, used to make assertions.
+
+```rust
+let [element] = "Hello" // Compile error, type mismatch
+let [element] = [1, 2] // Erlang runtime error
+let [element] = [1] // works
```

Comment:
It would be preferable to use `assert` for this runtime error. In future versions of Gleam a binding that can fail will not be accepted by the compiler if `let` is used.